scripts/map.js

```diff
@@ -182,13 +175,9 @@ function Map(display, __game) {
                 return true;
             } else if (typeof object.impassable === 'function') {
                 // the obstacle is impassable only in certain circumstances
-                try {
-                    return this._validateCallback(function () {
-                        return !object.impassable(__player, object);
-                    });
-                } catch (e) {
```

Comment:
This catch statement was dead code; `validateCallback` catches any errors the user may produce.